src/Components/Web/src/Forms/EditForm.cs

```diff
@@ -36,7 +37,12 @@ public EditForm()
         /// also supply <see cref="Model"/>, since the model value will be taken
         /// from the <see cref="EditContext.Model"/> property.
         /// </summary>
-        [Parameter] public EditContext? EditContext { get; set; }
+        [Parameter]
+        public EditContext? EditContext
+        {
+            get => _fixedEditContext;
+            set => _providedEditContext = value;
+        }
```

Comment:
I think this risks being a little weird, because I know we don't recommend people write to parameter properties directly, but if someone does this will be kind of bizarre. If you write a value to this property then try to read it back, you'll get a totally different value :)

Would it be possible instead to do something like:

```cs
bool _hasSetEditContextExplicitly;

[Parameter]
public EditContext? EditContext
{
    get => _fixedEditContext;
    set
    {
        _fixedEditContext = value;
        _hasSetEditContextExplicitly = true;
    }
}
```

... and then inside `OnParametersSet`:

```cs
if (_hasSetEditContextExplicitly && Model != null)
{
    throw "Don't set both"
}
else if (EditContext == null && Model == null)
{
   throw "Set at least one"
}
```

Sorry for not noticing this earlier on the original proposal!

Reply:
This is why I can't be left alone to review things after lunch 😄

Reply:
THis makes sense, I'll make the change